src/monitoring_service/database.py

```diff
@@ -143,6 +157,94 @@ def get_channel(self, token_network_address: str, channel_id: int) -> Optional[C
     def channel_count(self) -> int:
         return self.conn.execute("SELECT count(*) FROM channel").fetchone()[0]
 
+    def upsert_scheduled_event(self, event: ScheduledEvent) -> None:
+        contained_event: SubEvent = cast(SubEvent, event.event)
+        values = [
+            hex(event.trigger_block_number),
+            EVENT_TYPE_ID_MAP[type(contained_event)],
+            contained_event.token_network_address,
+            hex(contained_event.channel_identifier),
+            contained_event.non_closing_participant,
+        ]
+        with self.conn:
```

Comment:
Transaction should be outside the Database class or you will lose the possibility to group different writes into the same transaction. Even grouping some reads along with a single write can be necessary for consistency when having concurrent db access.

Apparently, we want one transaction be write in many cases and the transaction handling is getting tedious. We could us autocommit on our connection by setting `isolation_level = None` when creating the connection. See https://docs.python.org/3.7/library/sqlite3.html#controlling-transactions.